src/extensions/migrations/20240610GP-add-permissions-to-read-credit-additions-user-id.js

```diff
@@ -0,0 +1,71 @@
+const DIRECTUS_URL = process.env.DIRECTUS_URL;
+const ADMIN_ACCESS_TOKEN = process.env.ADMIN_ACCESS_TOKEN;
+const USER_ROLE_NAME = 'User';
+
+const COLLECTION_NAME = 'gp_credits_deductions';
+const FIELDS_TO_REMOVE = [];
+const FIELDS_TO_ADD = [ 'user_id' ];
+
+async function getUserRoleId () {
+	const URL = `${DIRECTUS_URL}/roles?filter[name][_eq]=${USER_ROLE_NAME}&access_token=${ADMIN_ACCESS_TOKEN}`;
+	const response = await fetch(URL).then((response) => {
+		if (!response.ok) {
+			throw new Error(`Fetch request failed. Status: ${response.status}`);
+		}
+
+		return response.json();
+	});
+	return response.data[0].id;
+}
```

Comment:
It seems like all these functions could be pulled out to a utils file and imported in each migration. Right now, it's:
 - lots of boilerplate,
 - hard to review because all the important parts are at the top in the constants, and the actual `up` function is just

```js
export async function up () {
	const roleId = await getUserRoleId();
	const { readPermissions } = await getUserPermissions(roleId);
	await patchReadPermissions(readPermissions);
}
```

...which doesn't say anything useful. It would be much better with:

```js
	const roleId = await getUserRoleId('User');
	const { readPermissions } = await getUserPermissions(roleId, 'gp_credits_deductions');
	await patchReadPermissions(readPermissions, ...);
```